AnimationGroup.lua

```diff
@@ -271,15 +291,26 @@ function AG:Fire(group, animation, signal)
     end
 
     -- We `bounce' if the boundary orders have no animations
-    if (not group.playing and shift and
-        (not group.finishing) and bouncing) then
-        group.reverse = not group.reverse
+    if (signal == 'Finished' and not group.playing and shift and
+        (not group.finishing) and (bouncing or repeating)) then
+        --group.shifted = true
+        if repeating then
+            group.order = -1
+        else
+            --group.order = group.order - (group.reverse and -1 or 1)
+            group.reverse = not group.reverse
+        end
+        group_func = group.handlers['OnLoop']
+        table.insert(args, group.reverse and 'REVERSE' or 'FORWARD')
+        if type(group_func) == 'function' then
+            group_func(group, unpack(args))
+        end
```

Comment:
```suggestion
    if (signal == 'Finished' and not group.playing and shift and
        (not group.finishing) and (bouncing or repeating)) then

        if repeating then
            group.order = -1
        else
            group.reverse = not group.reverse
        end
        group_func = group.handlers['OnLoop']
        table.insert(args, group.reverse and 'REVERSE' or 'FORWARD')
        if type(group_func) == 'function' then
            group_func(group, unpack(args))
        end
```

If you need these (comments) for later, please use the git stash :)